### ISSUE TITLE
Python kernel dies executing fc.mef.get_transform_fxn()

### ISSUE THREAD

**Issue by SynbioLucas** [taborlab/FlowCal#234]
When trying to fit bead data with the method `fc.mef.get_transform_fxn()`, the Python kernel dies for an unknown reason.

Code (run in Jupyter Notebook):
```
# coding: utf-8

# In[1]:

get_ipython().magic(u'matplotlib inline')

home = False
import numpy as np
import scipy as sp
import numpy.ma as ma
import matplotlib.pyplot as plt
import math, cmath, csv, sys, cPickle, os, os.path, gc
import lmfit as lm
import FlowCal as fc


# In[2]:

base_dir = os.path.abspath(os.path.join(os.path.abspath('__file__'), '..', '..'))
fig_dir = os.path.join(base_dir, 'Figures')
data_dir = os.path.join(base_dir, 'Flow Data')


# In[3]:

dates = ['20170117', '20170124']


# In[4]:

bead_dir = os.path.join(data_dir, dates[0], 'Data005.fcs') ## Bead data
bead_data = fc.io.FCSData(bead_dir)


# In[5]:

bead_MEFs = np.array([np.nan]+[792, 2079, 6588, 16471, 47497, 137049, 271647])
sc_channels = ['FSC', 'SSC']
fl_channels = ['FL1', 'FL2', 'FL3']


# In[ ]:

bead_data = fc.gate.start_end(bead_data, num_start=250, num_end=100)
#bead_data = fc.gate.high_low(bead_data, sc_channels)
gated_bead_data, _, gate_contour = fc.gate.density2d(data=bead_data,channels=sc_channels,
                                                     gate_fraction=0.4, full_output=True)


# In[ ]:

to_mef = fc.mef.get_transform_fxn(gated_bead_data, [bead_MEFs], clustering_channels=fl_channels,
                                  mef_channels=['FL1'], verbose=True, plot=True, plot_dir=None)
```

OS: Linux Mint 18.1
Packages installed: 
[linux_packages.txt](https://github.com/taborlab/FlowCal/files/751832/linux_packages.txt)
Working pacakges (produce no error on Windows 10): 
[windows_packages.txt](https://github.com/taborlab/FlowCal/files/751833/windows_packages.txt)

**Comment by castillohair** [comment]
Did you ever try to run this on a regular python script, as opposed to an ipython notebook? Also, can you include the relevant flow files? (I guess 'Data0005.fcs' would be enough).

btw, I'm currently using linux mint 18 with your package versions, and the example files run fine (with the fix for matplotlib 2.0.0). So this may be something specific to your flow file, or mint 18.1, or something defective with your installation.

**Comment by SynbioLucas** [comment]
> Did you ever try to run this on a regular python script, as opposed to an ipython notebook?

Running this in an IPython instance in the console yields the following message:

```
/home/lucas/anaconda2/lib/python2.7/site-packages/sklearn/utils/deprecation.py:52: DeprecationWarning: Class GMM is deprecated; The class GMM is deprecated in 0.18 and will be  removed in 0.20. Use class GaussianMixture instead.
  warnings.warn(msg, category=DeprecationWarning)
/home/lucas/anaconda2/lib/python2.7/site-packages/sklearn/utils/deprecation.py:70: DeprecationWarning: Function log_multivariate_normal_density is deprecated; The function log_multivariate_normal_density is deprecated in 0.18 and will be removed in 0.20.
  warnings.warn(msg, category=DeprecationWarning)
Intel MKL FATAL ERROR: Cannot load libmkl_avx2.so or libmkl_def.so.
```

Based on [this](https://github.com/BVLC/caffe/issues/3884), I ran `conda install mkl`. Running the code again yields:

```
/home/lucas/anaconda2/lib/python2.7/site-packages/sklearn/utils/deprecation.py:52: DeprecationWarning: Class GMM is deprecated; The class GMM is deprecated in 0.18 and will be  removed in 0.20. Use class GaussianMixture instead.
  warnings.warn(msg, category=DeprecationWarning)
/home/lucas/anaconda2/lib/python2.7/site-packages/sklearn/utils/deprecation.py:70: DeprecationWarning: Function log_multivariate_normal_density is deprecated; The function log_multivariate_normal_density is deprecated in 0.18 and will be removed in 0.20.
  warnings.warn(msg, category=DeprecationWarning)
/home/lucas/anaconda2/lib/python2.7/site-packages/sklearn/utils/deprecation.py:70: DeprecationWarning: Function log_multivariate_normal_density is deprecated; The function log_multivariate_normal_density is deprecated in 0.18 and will be removed in 0.20.
  warnings.warn(msg, category=DeprecationWarning)
/home/lucas/anaconda2/lib/python2.7/site-packages/sklearn/utils/deprecation.py:70: DeprecationWarning: Function log_multivariate_normal_density is deprecated; The function log_multivariate_normal_density is deprecated in 0.18 and will be removed in 0.20.
  warnings.warn(msg, category=DeprecationWarning)
/home/lucas/anaconda2/lib/python2.7/site-packages/sklearn/utils/deprecation.py:70: DeprecationWarning: Function log_multivariate_normal_density is deprecated; The function log_multivariate_normal_density is deprecated in 0.18 and will be removed in 0.20.
  warnings.warn(msg, category=DeprecationWarning)
Step 1: Clustering
  Number of populations to find: 8
  Percentage of events in each population:
    [ 11.2   13.85  12.87  13.72  11.93  12.22  12.39  11.82]
(FL1) Step 2: Population Statistic
  Fluorescence of each population (RFI):
    [ 102.  197.  197.  199.  484.  587.  483.  587.]
(FL1) Step 3: Population Selection
  3 populations selected.
  Fluorescence of selected populations (RFI):
    [ 197.  197.  199.]
  Fluorescence of selected populations (MEF):
    [  792.  2079.  6588.]
(FL1) Step 4: Standard Curve Fitting
  Parameters of bead fluorescence model:
    [  6.37e+00  -2.24e+01   7.67e+04]
```

This looks like progress, but  I'm concerned that FlowCal is only identifying 3 bead populations in my bead data file. :( 


> Also, can you include the relevant flow files? (I guess 'Data0005.fcs' would be enough).

Available here: [link](https://drive.google.com/open?id=0B2QoVlwqKtqgRWFhdmtVZHdUemM).

Just let me know if you have any other questions.

**Comment by castillohair** [comment]
I found the problem. Your code is only going to work with FlowCal 1.0.0 (which you are using in windows). Since FlowCal 1.1.0, transformation to arbitrary units is required as an intermediate step, and should be done before calculating the MEF transformation function and applying the transformation function to cell sample files. The reason why we did this is complicated, but the short story is that this was needed to improve compatibility with other flow cytometers.

Your code can be fixed by adding this line before calculating the MEF transformation function:

```python
bead_data = fc.transform.to_rfi(bead_data, sc_channels + fl_channels)
```

If you plan on writing code for FlowCal >= 1.1.0, using the above line right after you load an .fcs file should be enough.

Without this line, the following results:
![populations_fl1_data005 fcs](https://cloud.githubusercontent.com/assets/1529481/23314253/c4aa80ce-fa86-11e6-85c9-56b15f96f8af.png)


With this line, the following results:
![populations_fl1_data005 fcs](https://cloud.githubusercontent.com/assets/1529481/23314242/b8474254-fa86-11e6-84b2-5b134d579f12.png)

(plots look a little different because matplotlib2.)

Hope this helps. I'll close the issue now.
